localstack-core/localstack/services/lambda_/provider.py

```diff
@@ -1866,6 +1869,39 @@ def update_alias(
     # =======================================
     # ======= EVENT SOURCE MAPPINGS =========
     # =======================================
+    def check_service_resource_exists(self, service, resource_arn):
+        arn = parse_arn(resource_arn)
+        source_client = get_internal_client(resource_arn)
+        if service in ["sqs", "sqs-fifo"]:
+            try:
+                source_client.get_queue_url(QueueName=arn["resource"])
+            except ClientError as e:
+                if e.response["Error"]["QueryErrorCode"] == "QueueDoesNotExist":
```

Comment:
question: why use the `QueryErrorCode` here? Since the error message returned by Lambda returns the SQS error code as `AWS.SimpleQueueService.NonExistentQueue`?

Should we use the regular `Code` one, and also return that in the error message? As it is exactly what is returned by the SQS service, we might not need to hardcode it here, and directly return what SQS is returning.

Reply:
mostly because it was shorter tbh, but yes, reformatting this